batchflow/batch.py

```diff
@@ -528,24 +593,22 @@ def apply_transform(self, ix, func, *args, src=None, dst=None, p=None, use_self=
         ::
 
             apply_transform(make_masks_fn, src='images', dst='masks')
-            apply_transform(apply_mask, src=('images', 'masks'), dst='images', use_self=True)
-            apply_transform_all(rotate, src=['images', 'masks'], dst=['images', 'masks'], p=.2)
+            apply_transform(apply_mask, src=('images', 'masks'), dst='images')
+            FIXME apply_transform(rotate, src=['images', 'masks'], dst=['images', 'masks'], p=.2)
+            apply_transform(MyBatch.some_static_method, p=.5)
+            apply_transform(B.some_method, p=.5)
         """
-        dst = src if dst is None else dst
-
-        if not (isinstance(dst, str) or
-                (isinstance(dst, (list, tuple)) and np.all([isinstance(component, str) for component in dst]))):
-            raise TypeError("dst should be str or tuple or list of str")
-
-        p = 1 if p is None or np.random.binomial(1, p) else 0
+        target = target if target is not None else self.defaults['target']
+        init = init if init is not None else self.defaults['init']
+        post = post if post is not None else self.defaults['post']
+        src = src if src is not None else self.defaults['src']
+        dst = dst if dst is not None else self.defaults['dst']
```

Comment:
Code duplication. Can be solved by merging `defaults` and `kwargs` with the following unpacking.

Reply:
To achieve this, we need to move `target=None, init=None, post=None, src=None, dst=None, p=None` to `**kwargs` in a method header. Wouldn't that be too implicit? Like, the method header will look so: `def apply_transform(self, func, *args, **kwargs)`.